### PR TITLE
HADOOP-18853. Upgrades SDK version to 2.20.28 and restores multipart copy

### PR DESCRIPTION
**Pull request by ahmarsuhail**
### Description of PR

* Upgrades SDK Version so it has the new Java Async client with MPU support.
* Configures java async client with MPU threshold and part size
* Removes CRT dependency

### How was this patch tested?

Tested in eu-west-1 with `mvn -Dparallel-tests -DtestsThreadCount=16 clean verify`. 

Also ran checked output of ITestS3HugeFileArrayBlocks.test_100_renameHugeFile(), time taken to rename a 256MB file is at par with V1, around ~1s from my m4.2xlarge EC2.

### CODE REVIEW

**Comment by ahmarsuhail** [comment]
@steveloughran this updates the SDK version, so the Java async client has MPU. Since Multipart operations are in the java async client now, and not the TM, we could consider removing the TM in a follow up PR. We'd lose out on the transfer listener..but we don't really use any of the other additional functionality the TM provides us.

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java
Comment (steveloughran) [inline hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java:103]: is this going to build properly with unshaded artifacts? we've caused problems in the path (#2599) because of refs to .shaded classes.. the netty and client stuff are public/stable unshaded classes, correct?

**Review by steveloughran — APPROVED** [review]
Looks good though I want my #5872 patch in first -which is depending on approval from someone with commit rights.

please make sure jira and pr title states what library is being updated and what is it changing too, e.g

Upgrade AWS SDK to 2.19.12 and restore multipart copy

Once this PR is merged in the v2 line is feature complete w.r.t the v1 SDK code, isn't it? which means it'll be time to rebase the branch then merge to trunk, plus same for the 3.3 backport!

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 40s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  1s |  |  xmllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ feature-HADOOP-18073-s3a-sdk-upgrade Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 15s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  30m 59s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed  |
| +1 :green_heart: |  compile  |  17m 27s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  16m 28s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   4m 29s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed  |
| +1 :green_heart: |  mvnsite  |   1m 50s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed  |
| +1 :green_heart: |  javadoc  |   1m 34s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 30s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 52s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| -1 :x: |  spotbugs  |   1m 21s | [/branch-spotbugs-hadoop-tools_hadoop-aws-warnings.html](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5960/1/artifact/out/branch-spotbugs-hadoop-tools_hadoop-aws-warnings.html) |  hadoop-tools/hadoop-aws in feature-HADOOP-18073-s3a-sdk-upgrade has 1 extant spotbugs warnings.  |
| +1 :green_heart: |  shadedclient  |  34m 32s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 58s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 48s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 48s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  16m 48s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m  7s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  javac  |  16m  7s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m 19s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 48s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 31s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 40s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 47s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  34m 59s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 45s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 11s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 14s |  |  The patch does not generate ASF License warnings.  |
|  |   | 219m 39s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5960/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5960 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle |
| uname | Linux c70e7301253f 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | feature-HADOOP-18073-s3a-sdk-upgrade / 6a328d35ae466a04959b693a0785b0a0b29a07c2 |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5960/1/testReport/ |
| Max. process+thread count | 672 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5960/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by steveloughran** [comment]
@ahmarsuhail can you rebase and retest this, then i'll merge. once in we can rebase the whole feature branch, retest -and then merge to trunk!

**Comment by ahmarsuhail** [comment]
@steveloughran  - Rebased and tested, mostly all good. `ITestS3AHugeFilesNoMultipart` is failing, I would expect it to pass now that we have multipart copy, but this can be investigated separately. 

Also seeing failures in `ITestRoleDelegationInFilesystem>ITestSessionDelegationInFilesystem`

```
org.apache.hadoop.fs.s3a.auth.delegation.DelegationTokenIOException: Cannot issue S3A Role Delegation Tokens without full AWS credentials
	at org.apache.hadoop.fs.s3a.auth.delegation.RoleTokenBinding.lambda$createTokenIdentifier$0(RoleTokenBinding.java:146)
```

Fails on my EC2, but works ok on my mac. It's probably a config issue, but haven't seen this ever before so need to investigate. Again, don't think it has anything to do with this change. so feel like we should be good to merge. 

Once we're in trunk, I'll work on adding CSE, that's the only thing missing now.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 39s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  0s |  |  xmllint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ feature-HADOOP-18073-s3a-sdk-upgrade Compile Tests _ |
| +0 :ok: |  mvndep  |   7m 52s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  38m 28s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed  |
| +1 :green_heart: |  compile  |  17m  6s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  16m  3s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   4m 32s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed  |
| +1 :green_heart: |  mvnsite  |   1m 51s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed  |
| +1 :green_heart: |  javadoc  |   1m 33s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 41s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 53s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  34m 27s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 58s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 46s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 27s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  16m 27s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  16m  6s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  javac  |  16m  6s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m 18s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 49s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 32s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 41s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 46s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  34m 55s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 45s |  |  hadoop-project in the patch passed.  |
| +1 :green_heart: |  unit  |   3m 17s |  |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 12s |  |  The patch does not generate ASF License warnings.  |
|  |   | 219m 48s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5960/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5960 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle |
| uname | Linux 6d4c2e5e1397 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | feature-HADOOP-18073-s3a-sdk-upgrade / 54dcf48d4e2918ceb485f74dce644714669b204e |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5960/2/testReport/ |
| Max. process+thread count | 703 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-tools/hadoop-aws U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5960/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by steveloughran** [comment]
your ec2 vm is using IAM session credentials, isn't it? so it can't issue full credential DTs. probably worth having the test skip if that's the case

File: hadoop-project/pom.xml
Comment (steveloughran) [inline hadoop-project/pom.xml]: while you are here, how about renaming evenstream to eventstream; only just noticed this

**Review by steveloughran — CHANGES_REQUESTED** [review]
looks good.
needs matching changes in LICENSE-binary; update the bundle version and cut the crt

File: hadoop-project/pom.xml
Comment (ahmarsuhail) [reply hadoop-project/pom.xml]: done

**Review by steveloughran — APPROVED** [review]
LGTM
+1

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 44s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  xmllint  |   0m  1s |  |  xmllint was not available.  |
| +0 :ok: |  shelldocs  |   0m  1s |  |  Shelldocs was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ feature-HADOOP-18073-s3a-sdk-upgrade Compile Tests _ |
| +0 :ok: |  mvndep  |  15m  8s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  32m  1s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed  |
| +1 :green_heart: |  compile  |  17m 19s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  compile  |  16m  5s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   4m 24s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed  |
| +1 :green_heart: |  mvnsite  |  19m 26s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed  |
| +1 :green_heart: |  javadoc  |   8m 53s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 31s |  |  feature-HADOOP-18073-s3a-sdk-upgrade passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 24s |  |  branch/hadoop-project no spotbugs output file (spotbugsXml.xml)  |
| +1 :green_heart: |  shadedclient  |  61m 37s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   1m  1s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  31m 19s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m  3s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javac  |  17m  3s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  15m 58s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  javac  |  15m 58s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   4m 16s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |  15m 35s |  |  the patch passed  |
| +1 :green_heart: |  shellcheck  |   0m  0s |  |  No new issues.  |
| +1 :green_heart: |  javadoc  |   8m 38s |  |  the patch passed with JDK Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   7m 26s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +0 :ok: |  spotbugs  |   0m 24s |  |  hadoop-project has no data from spotbugs  |
| +1 :green_heart: |  shadedclient  |  62m 25s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| -1 :x: |  unit  | 784m 13s | [/patch-unit-root.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5960/3/artifact/out/patch-unit-root.txt) |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m 59s |  |  The patch does not generate ASF License warnings.  |
|  |   | 1145m 17s |  |  |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.yarn.server.applicationhistoryservice.webapp.TestAHSWebServices |
|   | hadoop.mapreduce.v2.TestMRJobsWithProfiler |
|   | hadoop.mapreduce.v2.TestMRJobs |
|   | hadoop.mapreduce.v2.TestUberAM |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5960/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5960 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient codespell detsecrets xmllint spotbugs checkstyle shellcheck shelldocs |
| uname | Linux e043f6811c5b 4.15.0-212-generic #223-Ubuntu SMP Tue May 23 13:09:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | feature-HADOOP-18073-s3a-sdk-upgrade / 9706ea8c494b283d47400848af35c7e7994ff09b |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20+8-post-Ubuntu-1ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5960/3/testReport/ |
| Max. process+thread count | 3938 (vs. ulimit of 5500) |
| modules | C: hadoop-project hadoop-tools/hadoop-aws . U: . |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5960/3/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 shellcheck=0.7.0 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**
